lib/VolSurface/Calibration/Equities.pm

```diff
@@ -0,0 +1,637 @@
+package VolSurface::Calibration::Equities;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.01';
+
+=head1 NAME
+
+VolSurface::Calibration::Equities
+
+=head1 DESCRIPTION
+
+This is a calibration model that is used in volatility surface calibration. This model checks whether a given volatility surface 
+conforms to some specific mathematical equations describing a volatility smile. If not, it will update the surface to fix it.
+
+=head1 VERSION
+
+Version 0.01
+
+=cut
+
+use Moose;
+
+use Date::Utility;
+use List::MoreUtils qw(pairwise first_index indexes);
+use List::Util qw(min max);
+use Math::Trig qw(tanh);
+use Try::Tiny;
+use Format::Util::Numbers qw(roundnear);
+
+has surface => (
+    is       => 'ro',
+    isa      => 'HashRef',
+    required => 1,
+);
+
+=head2 term_by_day
+
+Get all the terms in a surface in ascending order.
+
+=cut
+
+has term_by_day => (
+    is       => 'ro',
+    isa      => 'ArrayRef',
+    required => 1,
+);
+
+=head2 parameterization
+
+The parameterized (and, thus, smoothed) version of this surface.
+
+=cut
+
+has parameterization => (
+    is  => 'rw',
+    isa => 'Maybe[HashRef]',
+);
+
+=head2 smile_points
+
+The points across a smile.
+
+It can be delta points, moneyness points or any other points that we might have in the future.
+
+=cut
+
+has smile_points => (
+    is       => 'ro',
+    isa      => 'ArrayRef',
+    required => 1,
+);
+
+has _max_iteration => (
+    is      => 'ro',
+    isa     => 'Int',
+    default => 500,
+);
+
+has _tolerance_level => (
+    is      => 'ro',
+    isa     => 'Num',
+    default => 1e-05,
+);
+
+has _scale => (
+    is      => 'ro',
+    isa     => 'Num',
+    default => 10,
+);
+
+=head2 function_to_optimize
+
+The function that we want to optimize.
+
+=cut
+
+sub function_to_optimize {
+    my ($self, $params) = @_;
+
+    # this number is returned when the optimization is heading the wrong path
+    my $error = 1000;
+
+    my %params =
+        pairwise { our $a => our $b } @{$self->calibration_param_names},
+        @$params;
```

Comment:
Better:

```perl
my %params;
@params{$self->calibration_param_names} = @$params;
```

That's easier to understand for perl folks, I think, because they don't have to look up `pairwise`